recipes/android-ndk/all/test_package/conanfile.py

```diff
@@ -22,10 +23,13 @@ def build(self):
             cmake.build()
 
     def test(self):
-        if self.settings.os == "Windows":
-            self.run("ndk-build.cmd --version")
-        else:
-            self.run("ndk-build --version")
+        # INFO: can_run allows mac M1, but it does not work for the NDK
+        # https://github.com/android/ndk/issues/1299
+        if can_run(self) and not (self.settings.os == "Macos" and self.settings.arch == "armv8"):
+            if self.settings.os == "Windows":
+                self.run("ndk-build.cmd --version")
+            else:
+                self.run("ndk-build --version")
```

Comment:
```suggestion
        ndk_build = "ndk-build.cmd" if self.settings.os == "Windows" else "ndk-build"
        ndk_version = Version(self.tested_reference_str.split('/')[1])
        skip_run = platform.system() == "Darwin" and "arm" in platform.processor() and ndk_version < "r23c"
        if not skip_run:
            self.run(f"{ndk_build} --version", env="conanbuild")
        else:
            self.output.warning(f"Skipped running ndk-build on macOS Apple Silicon in arm64 mode, please use a newer"
                                 " version of the Android NDK")
```

Reply:
@sizeak I just tested @jcar87 suggestion locally and it works good, please, include it to your PR (I have no permission, needed to mark the checkbox to allow maintainers change the PR). Plus, you will need to add the follow lines in your imports:

```
from conan.tools.scm import Version
import platform
```

----

About the test_v1_package, I would ask deleting the entire folder and its content. We actually don't use any CMake target generated by cmake_find_package (legacy), we just build a simple cpp using the Android SDK.

Reply:
Or, you can just merge the PR https://github.com/glean-notes/conan-center-index/pull/1

Reply:
Done!